x/wasm/keeper/msg_dispatcher.go

```diff
@@ -140,12 +140,12 @@ func (d MessageDispatcher) DispatchSubmessages(ctx sdk.Context, contractAddr sdk
 
 		// we can ignore any result returned as there is nothing to do with the data
 		// and the events are already in the ctx.EventManager()
-		rData, err := d.keeper.reply(ctx, contractAddr, reply)
+		rspData, err := d.keeper.reply(ctx, contractAddr, reply)
 		switch {
 		case err != nil:
 			return nil, sdkerrors.Wrap(err, "reply")
-		case rData.Data != nil:
-			rsp = rData.Data
+		case rspData != nil:
```

Comment:
Hmmm... this may be subtly different. We need to double-check between `[]byte{}` and `nil`.
I would prefer `case len(rspData) > 0:`